### PR TITLE
[Rosetta-api] - Add vesting info

### PR DESCRIPTION
**Pull request by raynaudoe**
On this PR:
- Vesting info can be queried from `account/balance` using sub-accounts.
- Add `stx-lock` (tokens locking for staking) event parser.

### CODE REVIEW

**Comment by raynaudoe** [comment]
Tests are fixed now

File: src/api/rosetta-constants.ts [hunk 1]
```diff
@@ -12,8 +12,11 @@ export const RosettaConstants = {
   rosettaVersion: '1.4.6',
   symbol: 'STX',
   decimals: 6,
-  lockedBalance: 'LockedBalance',
-  spendableBalance: 'SpendableBalance',
+  StakedBalance: 'StakedBalance',
+  SpendableBalance: 'SpendableBalance',
+  VestingLockedBalance: 'VestingLockedBalance',
+  VestingUnlockedBalance: 'VestingUnlockedBalance',
+  VestingSchedule: 'VestingSchedule',
```
Comment (rllola) [inline src/api/rosetta-constants.ts:19]: I see that "LockedBalance" has disapeared. Does this mean that Vesting + Staked = Locked ?

Comment (rllola) [reply src/api/rosetta-constants.ts:19]: Just to be sure there is not a 3rd way to lock tokens. The other operation that I have in mind would be multisig balance but I guess it is considered spendable...

**Review by rllola — APPROVED** [review]

File: src/api/rosetta-constants.ts (hunk 1 above)
Comment (raynaudoe) [reply src/api/rosetta-constants.ts:19]: I've splitted the 'locked' notion of a balance into:
`StakedBalance`: locked balance due to Staking
`VestingLockedBalance`: locked balance due to vesting

So 'total locked' would be `StakedBalance` + `VestingLockedBalance`

**Pull request merged**